### PR TITLE
fix: add logging to sub process

### PR DESCRIPTION
**Pull request by orsagie**
- [x] Ready for review
- [x] Follows CONTRIBUTING rules
- [ ] Reviewed by Snyk internal team

#### What does this PR do?
Adds logging to sub process running sbt. To see the debug logs run the system-tests with `DEBUG=snyk-sbt-plugin npm run test-system`

#### How should this be manually tested?
`DEBUG=snyk-sbt-plugin npm run test-system`

#### What are the relevant tickets?
https://snyksec.atlassian.net/browse/BST-641

#### Screenshots
Screen Shot 2019-05-30 at 17 05 53

### CODE REVIEW

**Comment by lili2311** [comment]
Would be good to add a test:
- when snyk is run with `-d` we should debug output (testing this locally not seeing any extra output before we hang)

File: lib/sub-process.js [hunk 1]
```diff
@@ -1,28 +1,59 @@
...
+const childProcess = require('child_process');
+const debugModule = require('debug');
 
-module.exports.execute = function(command, args, options) {
-  var spawnOptions = {shell: true};
+// To enable debugging output, run the CLI as `DEBUG=snyk-sbt-plugin snyk ...`
+const debugLogging = debugModule('snyk-sbt-plugin');
+
+// TODO: this is not working as expected
+// const PROC_TIMEOUT = 300000; // 1 minute
```
Comment (lili2311) [inline lib/sub-process.js]: This is not working, I think maybe we need to set a timeout on CLI side? Or wrap this around the child process?

File: lib/sub-process.js [hunk 2]
```diff
@@ -1,28 +1,59 @@
...
+        debugLogging(str);
+      });
     });
 
-    proc.on('close', function(code) {
+    proc.on('close', (code) => {
       if (code !== 0) {
-        return reject(new Error(stdout || stderr));
+        return reject(new Error(
+          `>>> command: ${fullCommand} >>> exit code: ${code} ` +
```
Comment (lili2311) [inline lib/sub-process.js]: does this work with cli `-d`? Please can you share a screenshot of it working, I am not able to see any extra logging

Comment (orsagie) [reply lib/sub-process.js]: It doesn't work with `-d` It does work with `DEBUG=snyk-sbt-plugin snyk test`

Comment (lili2311) [reply lib/sub-process.js]: we should make it work with -d just like we have the gradle plugin, please take a look?

Comment (orsagie) [reply lib/sub-process.js]: This requires a change in the CLI

File: lib/sub-process.js (hunk 1 above)
Comment (orsagie) [reply lib/sub-process.js]: removed it for now. I will add another PR with timeouts

**Pull request merged**